### PR TITLE
FOIA-242: Document FOIA Advanced Auto Calculations module.

### PR DESCRIPTION
**Pull request by amanire**
- Add top level README documentation
- Standardize advanced calc JavaScript conventions 
- Apply Drupal coding standards to advanced calc JavaScript

### CODE REVIEW

File: docroot/modules/custom/foia_advcalc/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+FOIA Advanced Auto Calculations is a custom Drupal module so unlike contrib
+modules, the codebase is not installed via composer. Enable as you would
+normally enable a contributed Drupal module. Visit
+https://www.drupal.org/docs/8/extending-drupal-8/installing-drupal-8-modules#s-step-2-enable-the-module
+for further information.
+
+
+DOCUMENTATION
+-------------
+
```
Comment (bemarlan) [inline docroot/modules/custom/foia_advcalc/README.md]: Remove one of these lines

File: docroot/modules/custom/foia_advcalc/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+The FOIA Advanced Auto Calculations module attaches the `advcalc-fields.js` file
+to the "Annual FOIA Report Data" edit/add form.
+
```
Comment (bemarlan) [inline docroot/modules/custom/foia_advcalc/README.md]: Two line breaks before section titles.

File: docroot/modules/custom/foia_advcalc/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+`advcalc-fields.js` is organized as follows:
+
+ * Utility functions
+ * jQuery object calculation functions
+ * Annual Report field jQuery selector calculation bindings
+   * Initialize on load (for each binding)
+   * Calculate on change (for each binding)
+
+### jQuery selector reference
```
Comment (bemarlan) [inline docroot/modules/custom/foia_advcalc/README.md]: Two line breaks before section titles.

File: docroot/modules/custom/foia_advcalc/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+<input … name="field_foia_requests_va[1][subform][field_req_pend_start_yr][0][value]" …>
+```
+
+Where:
+
+ * `field_foia_requests_va` is the paragraph type machine name.
+ * `[1]` indicates that it's the second instance of the paragraph component.
+ * `field_req_pend_start_yr` is the field machine name.
+ * `[0]` indicates the first, and typically only, instance of the field on the
+paragraph instance.
```
Comment (bemarlan) [inline docroot/modules/custom/foia_advcalc/README.md]: Add 1 space before beginning second line.

File: docroot/modules/custom/foia_advcalc/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
+FOIA ADVANCED AUTO CALCULATIONS CUSTOM MODULE
+=============================================
+
+CONTENTS OF THIS FILE
+---------------------
+
+ * Introduction
+ * Requirements
+ * Installation
+ * Documentation
```
Comment (bemarlan) [inline docroot/modules/custom/foia_advcalc/README.md]: Add the three h3s under this section here as sub bullet.

File: docroot/modules/custom/foia_advcalc/README.md (hunk 1 above)
Comment (amanire) [reply docroot/modules/custom/foia_advcalc/README.md]: Done!

File: docroot/modules/custom/foia_advcalc/README.md (hunk 4 above)
Comment (amanire) [reply docroot/modules/custom/foia_advcalc/README.md]: Took me a second to understand this one, but I think I got it, i.e. 👉 hang that indent

**Pull request merged**